src/main/java/org/folio/rest/core/RestClient.java

```diff
@@ -18,60 +18,60 @@
-                    })
-                    .thenAccept(body -> {
-                        client.closeClient();
-                        if (logger.isDebugEnabled()) {
-                            logger.debug("The response body for GET {}: {}", endpoint, nonNull(body) ? body.encodePrettily() : null);
-                        }
-                        S responseEntity = body.mapTo(responseType);
-                        future.complete(responseEntity);
-                    })
-                    .exceptionally(t -> {
-                        client.closeClient();
-                        logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.GET, endpoint, requestContext), t);
-                        future.completeExceptionally(t.getCause());
-                        return null;
-                    });
-        } catch (Exception e) {
-          logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.GET, requestEntry.getBaseEndpoint(), requestContext), e);
+    try {
+      client
+        .request(HttpMethod.GET, endpoint, requestContext.getHeaders())
+        .thenApply(response -> {
+          if (logger.isDebugEnabled()) {
+            logger.debug("Validating response for GET {}", endpoint);
+          }
+          return verifyAndExtractBody(response);
+        })
+        .thenAccept(body -> {
           client.closeClient();
-          future.completeExceptionally(e);
-        }
-        return future;
+          if (logger.isDebugEnabled()) {
+            logger.debug("The response body for GET {}: {}", endpoint, nonNull(body) ? body.encodePrettily() : null);
+          }
+          S responseEntity = body.mapTo(responseType);
+          future.complete(responseEntity);
+        })
+        .exceptionally(t -> {
+          client.closeClient();
+          logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.GET, endpoint, requestContext), t);
+          future.completeExceptionally(t.getCause());
+          return null;
+        });
+    } catch (Exception e) {
+      logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.GET, requestEntry.getBaseEndpoint(), requestContext), e);
+      client.closeClient();
+      future.completeExceptionally(e);
     }
+    return future;
+  }
 
-    protected HttpClientInterface getHttpClient(Map<String, String> okapiHeaders) {
-        final String okapiURL = okapiHeaders.getOrDefault(RestClient.OKAPI_URL, "");
-        final String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(OKAPI_HEADER_TENANT));
+  protected HttpClientInterface getHttpClient(Map<String, String> okapiHeaders) {
+    final String okapiURL = okapiHeaders.getOrDefault(RestClient.OKAPI_URL, "");
+    final String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(OKAPI_HEADER_TENANT));
 
-        return HttpClientFactory.getHttpClient(okapiURL, tenantId);
+    return HttpClientFactory.getHttpClient(okapiURL, tenantId);
 
-    }
+  }
```

Comment:
rest client class is used 4 spaces but folio modules mostly use 2 space. so some adjustment has been done in this class